Pull Req: Always whitelist blackberry.event

rwmtse opened the pull request:
- Make blackberry.event always available even if the developer did not whitelist it
- Make sure blackberry.event gets loaded before any other extensions get loaded

Framework Pull Request: blackberry-webworks/BB10-WebWorks-Framework#37 <-- this pull request to be closed AFTER the framework pull request is closed, the Framework submodule needs to point to the right commit

Docs Pull Request: blackberry-webworks/WebWorks-API-Docs#118

rwmtse committed:
Automatically whitelist blackberry.event so that it is always available

rwmtse committed:
Make sure blackberry.event gets loaded before any other extensions

rwmtse committed:
Removed spyOn statement that is not needed

rwmtse committed:
Update Framework to point to refactored event code

nukulb commented on lib/config-parser.js:
```diff
@@ -44,6 +44,17 @@ function createAccessListObj(featuresArray, uri, allowSubDomain) {
     }
 
+    // always add free features to whitelist
+    fileManager.getFreeFeatures().forEach(function (feature) {
```
Can we rename getFreeFeatures to something more descriptive like getGlobalFeatures?

nukulb commented on lib/file-manager.js:
```diff
@@ -95,18 +100,39 @@ function getModulesArray(dest, files, baseDir) {
     return modulesList;
 }
 
+function isFreeModule(path) {
```
isGlobalModule might be better name?

nukulb commented on lib/file-manager.js:
```diff
@@ -95,18 +100,39 @@ function getModulesArray(dest, files, baseDir) {
 
+function isFreeModule(path) {
+    var extId = /(\w+)\/([a-z.]+)\/([a-z.]+)/.exec(path)[2];
+    return FREE_FEATURES.reduce(
```
GLOBAL_FEATURES

tracyli commented:
tested in windows: pause/resume works great while battery events is not working.

rwmtse committed:
1. Updated Framework to point to latest event code
2. Reverted frameworkModules.js change since the order for require.load does not really matter

tracyli commented:
tested in mac with http://mac-ci:9000/job/BB10-Webworks-Packager-next-always-whitelist-event/5/ and pauseresume and battery event work great.

issp commented:
Tested on windows: the battery events work as expected in build # :http://mac-ci:9000/job/BB10-Webworks-Packager-next-always-whitelist-event/5/

rwmtse committed:
Update Framework to point to latest event code

issp commented:
Tested on windows- the resume/pause and battery events work as expected in the device for build # :http://mac-ci:9000/job/BB10-Webworks-Packager-next-always-whitelist-event/6/

tracyli commented:
tested on mac with http://mac-ci:9000/job/BB10-Webworks-Packager-next-always-whitelist-event/6/ and no issue was found.

rwmtse committed:
Update Framework to include latest event code
